core/services/offchainreporting/discoverer_database.go

```diff
@@ -0,0 +1,62 @@
+package offchainreporting
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/lib/pq"
+	p2ppeer "github.com/libp2p/go-libp2p-core/peer"
+	"github.com/pkg/errors"
+	ocrnetworking "github.com/smartcontractkit/libocr/networking"
+	"go.uber.org/multierr"
+)
+
+var _ ocrnetworking.DiscovererDatabase = &DiscovererDatabase{}
+
+type DiscovererDatabase struct {
+	db     *sql.DB
+	peerID string
+}
+
+func NewDiscovererDatabase(db *sql.DB, peerID p2ppeer.ID) *DiscovererDatabase {
+	return &DiscovererDatabase{
+		db,
+		peerID.Pretty(),
+	}
+}
+
+// StoreAnnouncement has key-value-store semantics and stores a peerID (key) and an associated serialized
+// announcement (value).
+func (d *DiscovererDatabase) StoreAnnouncement(ctx context.Context, peerID string, ann []byte) error {
+	_, err := d.db.ExecContext(ctx, `
+INSERT INTO offchainreporting_discoverer_database (local_peer_id, remote_peer_id, ann, created_at, updated_at)
```

Comment:
nit: better name maybe just offchainreporting_discoverer_announcements since its just a table of announcements?